crates/solidity/outputs/cargo/crate/src/generated/language.rs

```diff
@@ -0,0 +1,9977 @@
+                if helper.handle_next_result(stream, result) {
+                    break;
+                }
+                let result =
+                    self.default_parse_token_with_trivia(stream, TokenKind::MinutesKeyword);
+                if helper.handle_next_result(stream, result) {
+                    break;
+                }
+                let result =
+                    self.default_parse_token_with_trivia(stream, TokenKind::SecondsKeyword);
+                if helper.handle_next_result(stream, result) {
+                    break;
+                }
+                let result = self.default_parse_token_with_trivia(stream, TokenKind::WeeksKeyword);
+                if helper.handle_next_result(stream, result) {
+                    break;
+                }
+                let result = self.default_parse_token_with_trivia(stream, TokenKind::WeiKeyword);
+                if helper.handle_next_result(stream, result) {
+                    break;
+                }
+                if !self.version_is_at_least_0_5_0 {
+                    let result =
+                        self.default_parse_token_with_trivia(stream, TokenKind::YearsKeyword);
+                    if helper.handle_next_result(stream, result) {
+                        break;
+                    }
+                }
+                if self.version_is_at_least_0_6_11 {
+                    let result =
+                        self.default_parse_token_with_trivia(stream, TokenKind::GweiKeyword);
+                    if helper.handle_next_result(stream, result) {
+                        break;
+                    }
+                }
+                if !self.version_is_at_least_0_7_0 {
+                    let result = {
+                        let mut helper = ChoiceHelper::new(stream);
+                        loop {
+                            let result = self
+                                .default_parse_token_with_trivia(stream, TokenKind::FinneyKeyword);
+                            if helper.handle_next_result(stream, result) {
+                                break;
+                            }
+                            let result = self
+                                .default_parse_token_with_trivia(stream, TokenKind::SzaboKeyword);
+                            if helper.handle_next_result(stream, result) {
+                                break;
+                            }
+                            break;
+                        }
+                        helper.result(stream)
+                    };
+                    if helper.handle_next_result(stream, result) {
+                        break;
+                    }
+                }
+                break;
+            }
+            helper.result(stream)
+        }
+    }
+
+    #[allow(unused_assignments, unused_parens)]
+    pub fn numeric_expression(&self, stream: &mut Stream) -> ParserResult {
```

Comment:
This PR exposes all of these functions and types to the public API. Is this by accident?
I suggest bringing back `pub(crate)` on all of them.

Reply:
I usually run `cargo doc --package slang_solidity --open --no-deps` locally to inspect the exposed public API if I made any changes.

We should probabably codegen/automate this, [once we have machine readable output](https://rust-lang.github.io/rfcs/2963-rustdoc-json.html) of this command.

Reply:
Yes, by accident when rewriting the codegen. Will fix.

Reply:
done

Reply:
Unfortunately ` slang_solidity::support` is still public.

Reply:
The problem here is that making it pub(crate) leaves the runtime crate with a lot of erroneous warnings, because of course the runtime doesn't include any *use* of the things in that crate. I'd rather fix that later by having some dummy file in the runtime that uses those things than put `#[allow(...)]` annotations on every method in those files.

Reply:
f2f: adding `allow()` on the entire crate instead.